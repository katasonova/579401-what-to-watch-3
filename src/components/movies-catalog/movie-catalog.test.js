import React from "react";
import renderer from 'react-test-renderer';
import MoviesCatalog from '../movies-catalog/movies-catalog.jsx';

const films = [{
  "id": 1,
  "name": `Harry Potter`,
  "poster_image": `img/harry-potter-poster.jpg`,
  "preview_image": `img/harry-potter.jpg`,
  "background_image": `img/harry-potter-bg.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://some-link`,
  "preview_video_link": `https://some-link`,
  "description": `Very interesting movie.`,
  "rating": 6.2,
  "scores_count": 125,
  "director": `Mr Nobody`,
  "starring": [`Dan Redcliff`, `Emma Watson`, `Helena Bonhem Carter`],
  "run_time": 1000,
  "genre": `Fantasy`,
  "released": 2000,
  "is_favorite": false
}, {
  "id": 2,
  "name": `Inception`,
  "poster_image": `img/inception-poster.jpg`,
  "preview_image": `img/inception.jpg`,
  "background_image": `img/inception-bg.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://some-link`,
  "preview_video_link": `https://some-link`,
  "description": `A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a C.E.O.`,
  "rating": 9.1,
  "scores_count": 456,
  "director": `C. Nolan`,
  "starring": [`Oscar`, `Famous actor1`, `Famous actor2`],
  "run_time": 124,
  "genre": `Thriller`,
  "released": 2010,
  "is_favorite": true
}];

it(`<SmallMovieCard /> is expected to render a movie card`, () => {
  const onCardTitleClick = jest.fn();

  const moviesList = renderer.create(<MoviesCatalog films={films} onCardTitleClick={onCardTitleClick}/>).toJSON();
  expect(moviesList).toMatchSnapshot();
});
