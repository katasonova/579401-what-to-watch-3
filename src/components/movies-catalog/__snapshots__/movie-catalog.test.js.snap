// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallMovieCard /> is expected to render a movie card 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Harry Potter"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Harry Potter
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Inception"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Inception
      </a>
    </h3>
  </article>
</div>
`;
